app/domain/errors.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+require 'util/error_class'
+
+unless defined? Errors::Authentication::AuthenticatorNotFound
+  # this wrapper is here so these classes will not be loaded by Rails
+  # auto-load. #TODO: fix this in a proper manner
+
+  module Errors
+    module Authentication
+
+      AuthenticatorNotFound = ::Util::ErrorClassWithCode.new(
+        msg: "'{0}' wasn't in the available authenticators",
```

Comment:
Can you please specify what is in the available authenticators add this " available authenticators are {1}"